ambry-store/src/main/java/com.github.ambry.store/StoreMetrics.java

```diff
@@ -18,131 +18,214 @@
+    private final Map<String, StoreLevelMetrics> storeLevelMetrics = new HashMap<>();
+    private final MetricRegistry registry;
+    public final Timer diskStartTime;
+    public final Counter diskStartFailure;
+
+    /**
+     * Create a {@link DiskLevelMetrics} object for handling metrics related to the stores on a certain disk.
+     * @param diskId the name of the disk.
+     */
+    public DiskLevelMetrics(String diskId, MetricRegistry registry) {
+      this.registry = registry;
+      String name = diskId + ".";
+      diskStartTime = registry.timer(MetricRegistry.name(DiskManager.class, name + "DiskStartTime"));
+      diskStartFailure = registry.counter(MetricRegistry.name(DiskManager.class, name + "DiskStartFailure"));
+    }
+
+    /**
+     * Create a {@link StoreLevelMetrics} object for handling metrics related to a specific store and make it
+     * accessible for lookup.
+     * @param storeId the name of the store.
+     * @return the {@link StoreLevelMetrics}
+     */
+    public StoreLevelMetrics createStoreLevelMetrics(String storeId) {
+      StoreLevelMetrics metrics = new StoreLevelMetrics(storeId, registry);
+      storeLevelMetrics.put(storeId, metrics);
+      return metrics;
+    }
+
+    /**
+     * @param storeId the name of the store to lookup the metrics object for.
+     * @return the {@link StoreLevelMetrics} for a disk name, or {@code null} if that disk could not be found.
+     */
+    public StoreLevelMetrics getStoreLevelMetrics(String storeId) {
+      return storeLevelMetrics.get(storeId);
+    }
   }
 
-  public void initializeHardDeleteMetric(final HardDeleter hardDeleter, final Log log) {
-    currentHardDeleteProgress = new Gauge<Long>() {
-      @Override
-      public Long getValue() {
-        return hardDeleter.getProgress();
-      }
-    };
-    registry.register(MetricRegistry.name(PersistentIndex.class, name + "CurrentHardDeleteProgress"),
-        currentHardDeleteProgress);
-
-    percentageHardDeleteCompleted = new Gauge<Double>() {
-      @Override
-      public Double getValue() {
-        return ((double) hardDeleter.getProgress() / log.getLogEndOffset()) * 100;
-      }
-    };
-    registry.register(MetricRegistry.name(Log.class, name + "PercentageHardDeleteCompleted"),
-        percentageHardDeleteCompleted);
-
-    hardDeleteThreadRunning = new Gauge<Long>() {
-      @Override
-      public Long getValue() {
-        return hardDeleter.isRunning() ? 1L : 0L;
-      }
-    };
-    registry.register(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteThreadRunning"),
-        hardDeleteThreadRunning);
-
-    hardDeleteCaughtUp = new Gauge<Long>() {
-      @Override
-      public Long getValue() {
-        return hardDeleter.isCaughtUp() ? 1L : 0L;
-      }
-    };
-    registry.register(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteCaughtUp"), hardDeleteCaughtUp);
+  /**
+   * Metrics for a specific store.
+   */
+  public static class StoreLevelMetrics {
```

Comment:
I added the `storeStartFailure` metric, but other than that, it should be the same as the old `StoreMetrics`.